packages/shared/src/components/buttons/Button.tsx

```diff
@@ -95,13 +95,16 @@ function ButtonComponent<TagName extends AllowedTags>(
         variant !== ButtonVariant.Option && 'justify-center font-bold',
         { iconOnly },
         iconOnly ? IconOnlySizeToClassName[size] : SizeToClassName[size],
+        iconPosition === ButtonIconPosition.Top && `flex-col`,
         !color && VariantToClassName[variant],
         VariantColorToClassName[variant]?.[color],
         className,
       )}
     >
       {icon &&
-        iconPosition === ButtonIconPosition.Left &&
+        new RegExp(
+          `(${ButtonIconPosition.Left}|${ButtonIconPosition.Top})`,
+        ).test(iconPosition) &&
```

Comment:
Why do we need this exactly?
Do you mean something like `[Left,Top].includes(iconPosition)`?

Reply:
I think regex is faster than `includes`, it also avoid having to create a new array -- this was just me being a little fancy 😅

Reply:
Inline regex can be really slow depending on browser. For example V8 has some good optimizations but Safari (iOS) and Firefox are much slower. If we are afraid of inline arrays we can just define const of `[Left,Top]` above component.

Reply:
Quick bench https://www.measurethat.net/Benchmarks/ShowResult/514328

Reply:
I also think personally just doing includes on array is a bit more readable.
(this specific regex is ok, but it can get very complicated ofcourse) 😅

Reply:
oh, nevermind then 👀